weave/flow/dataset.py

```diff
@@ -71,6 +75,64 @@ def to_pandas(self) -> "pd.DataFrame":
 
         return pd.DataFrame(self.rows)
 
+    def add_rows(self, rows: Iterable[dict]) -> "Dataset":
+        """Create a new dataset version by appending rows to the existing dataset.
+
+        This is useful for adding examples to large datasets without having to
+        load the entire dataset into memory.
+
+        Args:
+            rows: The rows to add to the dataset.
+
+        Returns:
+            The updated dataset.
+        """
+        client = get_weave_client()
+        if client is None:
+            raise ValueError("Weave is not initialized, call weave.init().")
+        if not isinstance(self.rows, WeaveTable) or not self.rows.table_ref:
+            raise TypeError(
```

Comment:
we can probably try to unit test each of these errors, free testcov

Reply:
I added tests for the no client and unpublished dataset error cases